### PR TITLE
ARROW-1265: [Plasma] Clean up all resources on SIGTERM to keep valgrind output clean

### PR DESCRIPTION
**Pull request by wesm**
@pcmoritz this might be a little bit OCD since all of the valgrind warnings were for memory that was still reachable, but let me know what you think.

### CODE REVIEW

File: cpp/src/plasma/events.h [hunk 1]
```diff
@@ -83,11 +83,14 @@ class EventLoop {
   /// @return The ae.c error code. TODO(pcm): needs to be standardized
   int remove_timer(int64_t timer_id);
 
-  /// Run the event loop.
+  /// \brief Run the event loop.
   ///
   /// @return Void.
   void run();
```
Comment (pcmoritz) [inline cpp/src/plasma/events.h]: Consider renaming this to EventLoop::Start for overall consistency with EventLoop::Stop

**Comment by pcmoritz** [comment]
Except for the small inconsistency, this looks good to me!

File: cpp/src/plasma/events.h (hunk 1 above)
Comment (wesm) [inline cpp/src/plasma/events.h]: Sounds good. I made a clean sweep over the rest of the EventLoop methods for consistency

**Comment by wesm** [comment]
Appveyor build running here, will merge when complete: https://ci.appveyor.com/project/wesm/arrow/build/1.0.811

**Pull request closed without merging**